Sources/TUSKit/TUSAPI.swift

```diff
@@ -38,6 +40,43 @@ final class TUSAPI {
         self.session = session
     }
     
+    @discardableResult
+    func serverInfo(server: URL, completion: @escaping (Result<TusServerInfo, TUSAPIError>) -> Void) -> URLSessionDataTask {
+        let request = makeRequest(url: server, method: .options, headers: [:])
+        let task = session.dataTask(request: request) { result in
+            processResult(completion: completion) {
+                let (_, response) =  try result.get()
+                
+                guard response.statusCode == 200 || response.statusCode == 204 else {
+                    throw TUSAPIError.couldNotFetchServerInfo
+                }
+                
+                var supportedAlgorithms: [String] = []
+                if let algorithms = response.allHeaderFields[caseInsensitive: "tus-checksum-algorithm"] as? String {
+                    supportedAlgorithms = algorithms.components(separatedBy: ",")
+                }
+                var supportedVersions: [String] = []
+                if let tusVersions = response.allHeaderFields[caseInsensitive: "tus-version"] as? String {
+                    supportedVersions = tusVersions.components(separatedBy: ",")
+                }
+                var maxSize: Int?
+                if let maxSizeStr = response.allHeaderFields[caseInsensitive: "tus-max-size"] as? String {
+                    maxSize = Int(maxSizeStr)
+                }
+                let version = response.allHeaderFields[caseInsensitive: "tus-resumable"] as? String ?? ""
+                
+                var extensions: [TUSProtocolExtension] = []
+                if let tusExtension = response.allHeaderFields[caseInsensitive: "tus-extension"] as? String {
+                    extensions = tusExtension.components(separatedBy: ",").map { TUSProtocolExtension(rawValue: $0) ?? .creation }
```

Comment:
What is the purpose of `?? .creation`? Does it assume that this extension is supported by default if no header is present? That is not always the case.